### DISCUSSION TITLE
"Real" maximize shortcut with stage manager (ignorin it)

### DISCUSSION THREAD

**Discussion by esseti** [rxhanson/Rectangle#1333] [General]
Hello,
I do use the maximize feature that makes the window full size keeping the space for stage manager. 
But sometimes i find myself in the need to have the stage manager "covered" by the window.[ I do so by double clicking the title bar of the window and it works perfectly.](https://support.apple.com/en-gb/guide/mac-help/mchlp2469/mac#:~:text=Maximise%20a%20window,Dock%20settings).) 

Is there a way to be able to do so?
i don't want to:
- remove the stage manager gap in the rectangle app, there are cases where the full screen with stage manager space is needed
- i don't want the window to be a full screen as it is by clicking the top left green button. (

**Comment by rxhanson** [comment]
Thanks for the feedback. I don't have plans to add this into Rectangle myself, but would be fine merging a good pull request for it as a terminal command enabled setting.

It does exist as a hidden feature of [Rectangle Pro](https://rectangleapp.com/pro) (paid), where repeating the maximize command will toggle between the two sizes. It is enabled with the following Terminal command followed by an app restart:

`defaults write com.knollsoft.Hookshot repeatedMaximizeMode -int 5`

**Comment by nertzy** [comment]
You can also:
- Option-click the top left green button
- select Window -> Zoom in the menu bar. 

You could attach a keyboard shortcut to Window -> Zoom if you want:

System Settings -> Keyboard -> Keyboard Shortcuts -> App Shortcuts -> Add a shortcut named "Zoom".
